extract-wos-article.py

```diff
@@ -23,7 +30,54 @@
     "Current Publisher": "publisher",
 }
 
-YEAR_REGEX = re.compile(".*([0-9]{4})")
+REGEX = {
+    "year": re.compile(".*([0-9]{4})"),
+    "wos_id_number": re.compile("([0-9]{15})"),
+}
+CITING_SUMMARY_URL_PREFIX = "https://www.webofscience.com/wos/woscc/citing-summary/"
+
+
+async def extract_citing_summary(page: Page, article_id: str):
+    url = CITING_SUMMARY_URL_PREFIX + article_id
+    await page.goto(url, wait_until="networkidle")
+
+    curr_page = await page.query_selector(CITING_SELECTORS["current_page"])
+    curr_page_text = await curr_page.input_value()
+    curr_page_num = int(curr_page_text.strip())
+
+    final_page = await page.query_selector(CITING_SELECTORS["final_page"])
+    final_page_text = await final_page.inner_html()
+    final_page_num = int(final_page_text.strip())
+
+    articles = []
+    while True:
```

Comment:
This feels a bit dangerous -- if the breaking condition never happens for some reason, it essentially ends up being an infinite loop.

Do you think it might make sense to add a maximal number of iterations with a sensible default here? Something like a 1000 would certainly work :slightly_smiling_face:

Reply:
@mrshu i guess this does the trick https://github.com/fmfi-svt/steel-wok/pull/4/commits/2374fa196b1d3a11e1426eb794021c9a16c7becb

Reply:
Nice, I like that!